homeassistant/components/mastodon/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""Define an object to manage fetching Mastodon data."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Any
+
+from mastodon import Mastodon
+from mastodon.Mastodon import MastodonError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER
+
+
+@dataclass
+class MastodonData:
+    """Mastodon data type."""
+
+    client: Mastodon
+    instance: dict
+    account: dict
+    coordinator: MastodonCoordinator
+
+
+type MastodonConfigEntry = ConfigEntry[MastodonData]
```

Comment:
Theoretically you can move this to init and then have a
```py
from __future__ import annotations

if TYPE_CHECKING:
    from . import MastodonConfigEntry
```
But it's opinionated, we don't have strict rules where this should live but I personally prefer the init